lib/ansible/cli/arguments/option_helpers.py

```diff
@@ -164,22 +164,23 @@ def _gitinfo():
 def version(prog=None):
     """ return ansible version """
     if prog:
-        result = " ".join((prog, __version__))
+        result = u" ".join((prog, __version__))
     else:
         result = __version__
 
     gitinfo = _gitinfo()
     if gitinfo:
-        result = result + " {0}".format(gitinfo)
-    result += "\n  config file = %s" % C.CONFIG_FILE
+        result = result + u" {0}".format(gitinfo)
+    result += u"\n  config file = %s" % to_text(C.CONFIG_FILE)
     if C.DEFAULT_MODULE_PATH is None:
-        cpath = "Default w/o overrides"
+        cpath = u"Default w/o overrides"
     else:
-        cpath = C.DEFAULT_MODULE_PATH
-    result = result + "\n  configured module search path = %s" % cpath
-    result = result + "\n  ansible python module location = %s" % ':'.join(ansible.__path__)
-    result = result + "\n  executable location = %s" % sys.argv[0]
-    result = result + "\n  python version = %s" % ''.join(sys.version.splitlines())
+        cpath = u":".join(C.DEFAULT_MODULE_PATH)
+    python_module_location = u":".join(map(to_text, ansible.__path__))
+    result = result + u"\n  configured module search path = %s" % cpath
+    result = result + u"\n  ansible python module location = %s" % python_module_location
+    result = result + u"\n  executable location = %s" % to_text(sys.argv[0])
+    result = result + u"\n  python version = %s" % ''.join(sys.version.splitlines())
```

Comment:
Should we have a `u` on the join too?  Also, does `sys.version.splitlines()` produce text? Should we be explicit there if not?

Reply:
Thanks, I completely missed the last `join`. Yes, it needs `u` and to be converted to text. Also, just noticed that `__version__` needs to be converted as well.

Reply:
Probably does not make sense to convert `__version__`.